src/api/app/models/repository_architecture.rb

```diff
@@ -1,11 +1,26 @@
 class RepositoryArchitecture < ApplicationRecord
+  serialize :required_checks, Array
   belongs_to :repository,   inverse_of: :repository_architectures
   belongs_to :architecture, inverse_of: :repository_architectures
 
   acts_as_list scope: [:repository_id], top_of_list: 0
 
   validates :repository, :architecture, :position, presence: true
   validates :repository, uniqueness: { scope: :architecture }
+
+  has_many :status_reports, as: :checkable, class_name: 'Status::Report', dependent: :destroy do
+    def for_uuid(uuid)
+      where(status_reports: { uuid: uuid })
+    end
+
+    def latest
+      for_uuid(proxy_association.owner.build_id).first
+    end
+  end
+
+  def build_id
+    Backend::Api::Build::Repository.build_id(repository.project.name, repository.name, architecture.name)
```

Comment:
Question: Does it have a build_id? I thought this concept only exists for published repositories?

Reply:
Since 8f304cf403538cefcf95a6691cf01df2416187cf it does